modules/raft/src/main/java/org/apache/ignite/internal/raft/ExecutorInclinedRaftCommandRunner.java

```diff
@@ -46,4 +46,8 @@ public <R> CompletableFuture<R> run(Command cmd) {
 
         return future.thenApplyAsync(identity(), completionExecutor);
     }
+
+    public RaftCommandRunner decoratedCommandRunner() {
```

Comment:
What's the purpose of this decorator?

Reply:
✅ A good question to revise for me. So:
1. The one of primary goals was to make a hole for `RaftGroupManager` from `ReplicaListener` through `Replica` and `ReplicaManager` to `TableManager`.
2. Then, I should have a chain of `.raftClien()` among that classes.
3. There is a problem: `RaftClient` has raft client as `RaftCommandRunner` instead `RaftGroupManager`. And the problem is that command runner interface is appliable for both raft client and for the discussed command runner that is a decorator. But the last class isn't exactly raft client. Then, there should be a workaround for `raftClient` implementation in `ReplicaListener`:
```java
    @Override
    public RaftCommandRunner raftClient() {
        if (raftClient instanceof ExecutorInclinedRaftCommandRunner) {
            return ((ExecutorInclinedRaftCommandRunner) raftClient).decoratedCommandRunner();
        }
        return raftClient;
    }
```